File: CSharp/167-TwoSum2InputArraySorted/167TwoSum2.cs
// TWO POINTERS
// the arary is already sorted. so min pointer starts at the left (low side) of array, min starts at high side of array.
// if sum is less than target move min pointer up. array is already sorting so this will increase the sum.
// if the sum is more than target move the max pointer left to decrease the sum. 
// it's a 1-indexed array instead of the usual 0-indexed so we need to add +1

// Time Complexity: O(n) (Analyze Complexity Confirmed)
// You have to go through each of the numbers in the int array

// Space Complexity: O(1) (Analyze Complexity Confirmed)
// We are just traversing the int array that is passed into the function, no extra memory is needed. 

public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int min = 0, max = numbers.Length - 1;
        while ((numbers[min] + numbers[max]) != target)
            if((numbers[min]+ numbers [max]) > target) max--;
            else min++;
        return new int[]{min + 1, max +1};
    }
}

File: Javascript/167-TwoSum2.InputArrayIsSorted/167TwoSum2.js
/* Two Pointers
Time Complexity: O(n)
With two pointers you only have to go over the array once starting at the ends. 
the pointers never cross.

Space Complexity: O(1)
we're just going over the numbers array given to us
same amount of pointers no matter the size of the array
*/
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
  let left = 0;
  let right = numbers.length - 1;
  while (left < right) {
    let total = numbers[left] + numbers[right];
    if (total === target) return [left + 1, right + 1];
    else if (total > target) right--;
    else left++;
  }
  throw new Error("No solution found for twoSum");
};
